app/services/notification_subscriptions/subscribe.rb

```diff
@@ -0,0 +1,91 @@
+module NotificationSubscriptions
+  class Subscribe
+    attr_reader :current_user, :permitted_params
+
+    def self.call(...)
+      new(...).call
+    end
+
+    def initialize(current_user, permitted_params)
+      @current_user = current_user
+      @permitted_params = permitted_params
+    end
+
+    def call
+      subscribe_subscription
+    end
+
+    def subscribe_subscription
+      create_subscription
+    end
+
+    private
+
+    def comment
+      @comment ||= Comment.find(permitted_params[:comment_id]) if permitted_params[:comment_id].present?
+    end
+
+    def article
+      @article ||= Article.find(permitted_params[:article_id]) if permitted_params[:article_id].present?
+    end
+
+    def comment_article
+      @comment_article ||= comment && comment.ancestry.nil? ? comment.commentable : nil
+    end
+
+    def notifiable_type
+      @notifiable_type ||= determine_notifiable_type
+    end
+
+    def notifiable
+      @notifiable ||= determine_notifiable
+    end
+
+    def create_subscription
+      subscription = NotificationSubscription.new(
+        user: current_user,
+        config: subscription_config,
+        notifiable: notifiable,
+        notifiable_type: notifiable_type,
+      )
+
+      if subscription.save
+        { updated: true, notification: subscription.to_json }
+      else
+        { errors: subscription.errors_as_sentence }
+      end
+    end
+
+    def determine_notifiable_type
```

Comment:
Minor: `determine_notifiable_type` repeats the logic from `determine_notifiable` and isn't necessary. (Rails' polymorphic associations require an explicit `_type` only when setting `_id` explicitly, so you'd only need `notifiable_type` if you were setting `notifiable_id`, but since you're setting `notifiable` then the `_type` is derivable.)